include/boost/compute/utility/program_cache.hpp

```diff
@@ -82,6 +82,35 @@ class program_cache : boost::noncopyable
         m_cache.clear();
     }
 
+    /// Clears persistent cache by deleting the boost_compute folder
+    static void clear_persistent_cache()
+    {
+#ifdef BOOST_COMPUTE_USE_OFFLINE_CACHE
+        // Path delimiter symbol for the current OS.
+        static const std::string delim = boost::filesystem::path("/").make_preferred().string();
+
+        // Path to appdata folder.
+#ifdef WIN32
+        static const std::string appdata = detail::getenv("APPDATA") 
+            + delim + "boost_compute";
+#else
+        static const std::string appdata = detail::getenv("HOME")
+            + delim + ".boost_compute";
+#endif
+        try
+        {
+            if(boost::filesystem::exists(appdata))
```

Comment:
I don't think you need both an existence check and a try/catch block. I'd lean towards just using the if(exists(appdata)) condition and then removing the try/catch.